CMakeLists.txt

```diff
@@ -41,20 +41,7 @@ if (MSVC)
 	set(objc_LINK_FLAGS "/DEBUG /INCREMENTAL:NO ${objc_LINK_FLAGS}")
 endif()
 
-# Get Architecture without relying on CMake
-try_compile(
-    COMPILE_SUCCESS
-    ${CMAKE_BINARY_DIR}
-    ${CMAKE_SOURCE_DIR}/CMake/detect_arch.c
-    OUTPUT_VARIABLE COMPILE_OUTPUT
-)
-
-if(NOT COMPILE_SUCCESS)
-    string(REGEX MATCH "(aarch64|arm|i386|x86_64|powerpc64|powerpc|unknown)" ARCHITECTURE ${COMPILE_OUTPUT})
-endif()
-
-set(ARCHITECTURE ${ARCHITECTURE} CACHE STRING "Architecture Type")
-message(STATUS "Architecture: ${ARCHITECTURE}")
+message(STATUS "Architecture as detected by CMake: ${CMAKE_SYSTEM_PROCESSOR}")
```

Comment:
I can’t remember why this was needed, are you sure it no longer is?

Reply:
There is still an issue when using CMake x86_64 on Windows on ARM64, as CMake reports AMD64 while using an ARM64 compiler.

The problem with this hack is that it searches for *any* occurrence in the output of the compiler. Other paths (like `/c/Program Files/python3-arm/`) are erroneously selected.

I think we should stick to the standard `CMAKE_SYSTEM_PROCESSOR` variable, but put a warning in `README.md` or `INSTALL`. CMake is not willing to fix this broken behaviour on Windows.